ci/jenkins/utils.sh

```diff
@@ -32,6 +32,8 @@ function check_and_cleanup_docker_build_cache() {
 }
 
 function check_and_upgrade_golang() {
+    echo "====== Clean up Golang cache ======"
+    go clean -cache -modcache -testcache || true
```

Comment:
Met the disk space issue again, I checked the Jenkins Node, the cache will be over 1G quite soon, so add a step to clean up cache unconditionally here. @XinShuYang @antoninbas can you take another look? Thanks.

```
root@antrea-kind-testbed:/var/lib/jenkins# du -sh $(go env GOCACHE)
1.9G	/var/lib/jenkins//.cache/go-build
root@antrea-kind-testbed:/var/lib/jenkins# du -sh $(go env GOMODCACHE)
2.1G	/var/lib/jenkins/go/pkg/mod
```